cli/api/commands/compile.ts

```diff
@@ -11,16 +13,26 @@ export class CompilationTimeoutError extends Error {}
 export async function compile(
   compileConfig: dataform.ICompileConfig = {}
 ): Promise<dataform.CompiledGraph> {
-  // Resolve the path in case it hasn't been resolved already.
-  path.resolve(compileConfig.projectDir);
+  let compiledGraph = dataform.CompiledGraph.create();
+  let workflowSettingsDataformCoreVersion = "";
+  try {
+    workflowSettingsDataformCoreVersion = await runInstallIfWorkflowSettingsDataformCoreVersion(
+      compileConfig.projectDir
```

Comment:
I don't think you should be doing _any_ filesystem mutations inside the project directory. If the process dies, cleanup will not happen. Even if we could guarantee that cleanup would happen, it would still not be great from a filesystem-watching perspective.

I think this should all be happening in /tmp.